packages/common/src/types/chainClients.ts

```diff
@@ -0,0 +1,65 @@
+import { RequireAtLeastOne } from "type-fest";
+import { BlockHeader } from "./block";
+import { Box } from "./boxes";
+import { HexString } from "./common";
+import { SignedTransaction, TransactionId, UnsignedTransaction } from "./transactions";
+
+export type QueryBase = {
+  take?: number;
+  skip?: number;
+};
+
+export type BoxQuery<W extends BoxWhere> = {
+  where: RequireAtLeastOne<W>;
+
+  /**
+   * Determines if should include unspent boxes from the mempool.
+   * @default true
+   */
+  includeUnconfirmed?: boolean;
+} & QueryBase;
+
+export type BoxWhere = {
+  /** Base16-encoded BoxId */
+  boxId?: HexString;
+
+  /** Base16-encoded ErgoTree or Base58-encoded address */
+  contract?: HexString;
+
+  /** Base16-encoded contract template */
+  template?: HexString;
+
+  /**  Base16-encoded TokenId */
+  tokenId?: HexString;
+};
+
+export type ChainClientBox = Box<bigint> & {
+  confirmed: boolean;
+};
+
+export interface IChainDataClient<B extends BoxWhere> {
+  /**
+   * Get unspent boxes from the blockchain.
+   */
+  getUnspentBoxes(query: BoxQuery<B>): Promise<ChainClientBox[]>;
+
+  /**
+   * Get the last `n` block headers from the blockchain.
+   */
+  getLastHeaders(count: number): Promise<BlockHeader[]>;
+
+  /**
+   * Check for transaction validity without broadcasting it to the network.
+   */
+  checkTransaction(transaction: SignedTransaction): Promise<boolean>;
```

Comment:
One small suggestion, since the return value is Boolean, wouldn't it be better if we call the function something like "isTransactionValid"?

Reply:
I believe the `is*` pattern is best suited for quick checks/validations. As this method is a potentially long running operation, it's best to keep `checkTransaction` to reflect that intent.